jerry-core/jerry.c

```diff
@@ -591,6 +591,47 @@ jerry_get_array_index_value (jerry_object_t *array_obj_p, /**< array object */
 } /* jerry_get_array_index_value */
 
 /**
+ * Get length of an array object
+ *
+ * @return length of the given array
+ */
+uint32_t
+jerry_get_array_length (const jerry_object_t *object_p)
+{
+  jerry_assert_api_available ();
+
+  JERRY_ASSERT (jerry_is_array (object_p));
```

Comment:
Fail-safe API? Might be a good idea, but all functions must follow this layout, which have perf cost.

E.g:
    if (is_number(value))
      val = get_number(value); // Adding an is_number check here slows down the code.

I am not against fail-safe API, but I would like to hear others opinion.